pkg/policy/api/egress.go

```diff
@@ -255,3 +345,28 @@ func (e *EgressRule) CreateDerivative(ctx context.Context) (*EgressRule, error)
 	e.SetAggregatedSelectors()
 	return newRule, nil
 }
+
+// CreateDerivative will return a new rule based on the data gathered by the
+// rules that creates a new derivative policy.
+// In the case of ToGroups will call outside using the groups callback and this
+// function can take a bit of time.
+func (e *EgressDenyRule) CreateDerivative(ctx context.Context) (*EgressDenyRule, error) {
+	newRule := e.DeepCopy()
+	if !e.RequiresDerivative() {
+		return newRule, nil
+	}
+	newRule.ToCIDRSet = CIDRRuleSlice{}
```

Comment:
nit: preallocate the slice

```suggestion
	newRule.ToCIDRSet = make(CIDRRuleSlice, 0, len(e.ToGroups)
```